site/public/ts/shared/utils/Exporter.ts

```diff
@@ -0,0 +1,100 @@
+import {XMLWriter} from "../../../../../app/core/ts/utils/io/xml/XMLWriter";
+import {DigitalCircuitDesigner} from "digital/models/DigitalCircuitDesigner";
+import {DigitalCircuitView} from "site/digital/views/DigitalCircuitView";
+import {V} from "Vector";
+import {Browser} from "core/utils/Browser";
+
+// Returns a 256x256 canvas on which the given circuit is centered
+//! No idea what happens if the designer is empty, divide-by-zero seems plausible
+//? how empty designers should be handled -- return empty canvas?
+function RenderCircuit(designer: DigitalCircuitDesigner): HTMLCanvasElement {
+    // 256 is arbitrary, could increase or decrease for more detail or less memory respectively
+    const THUMBNAIL_SIZE = 256;
+
+    // Find bounding box of the circuit
+    let xmin = Infinity;
+    let ymin = Infinity;
+    let xmax = -Infinity;
+    let ymax = -Infinity;
+    for (let c of designer.getObjects()) {
+        const cullbox = c.getCullBox();
+        const max = cullbox.getTopRight();
+        const min = cullbox.getBottomLeft();
+        xmin = Math.min(min.x, xmin);
+        ymin = Math.min(min.y, ymin);
+        xmax = Math.max(max.x, xmax);
+        ymax = Math.max(max.y, ymax);
+    }
```

Comment:
So there is not currently a way to get an axis-aligned bounding box for a component?

Reply:
There is, but the process of actually determining if an axis-aligned box is in the screen versus just a normal box is extremely expensive (since it's done every frame if zooming or panning) and kind of defeats the purpose of cull-boxes as an optimization (since cull-boxes only change when the component moves/changes)